### PR TITLE
add unified args number check

### PR DESCRIPTION
**Pull request by Chyroc**
## Description

* add `MinArgs` and `MaxArgs` field to `Command` struct
* and use it to check command args count in `Execute`, not in every command
* add valid: `del` command should have at least one arg

### CODE REVIEW

**Comment by Chyroc** [comment]
well pass ci when move from go get to dep

**Comment by kasvith** [comment]
What if we tried to add a MinArgs and MaxArgs field ? When we can use infinite num args we would set em to -1

**Comment by Chyroc** [comment]
done

**Comment by kasvith** [comment]
you need to run gofmt on each files

**Comment by Chyroc** [comment]
what is the fix follows mean, sorry for my bad english

File: internal/arch/command-table.go [hunk 1]
```diff
@@ -35,21 +35,23 @@ type CommandFunc func(*db.DB, []string) *protcl.Message
...
 	ModifyKeySpace bool
 	Fn             CommandFunc
+	MinArgs        int
+	MaxArgs        int
 }
 
 var CommandTable = map[string]Command{
 	// server
-	"ping": {ModifyKeySpace: false, Fn: cmds.Ping},
+	"ping": {ModifyKeySpace: false, Fn: cmds.Ping, MinArgs: -1, MaxArgs: 1},
```
Comment (kasvith) [inline internal/arch/command-table.go]: ping can have minimum 0 args and exactly 1 arg. So `MinArgs: 0` should be here

**Comment by kasvith** [comment]
@Chyroc sorry i was on mobile, review was not added below comment, now you can see it

**Comment by kasvith** [comment]
please run `go fmt` on each files

File: internal/arch/command-table.go (hunk 1 above)
Comment (Chyroc) [reply internal/arch/command-table.go]: done

**Comment by Chyroc** [comment]
i have run `make fmt` before i commit, so what file have not be formated

File: internal/arch/command-table.go [hunk 2]
```diff
@@ -70,5 +72,9 @@ func (DBCommand) Execute(db *db.DB, cmd string, args []string) *protcl.Message {
 		return protcl.NewMessage(nil, err)
 	}
 
+	if argsLen := len(args); (command.MinArgs != -1 && argsLen < command.MinArgs) || (command.MaxArgs != -1 && argsLen > command.MaxArgs) {
```
Comment (kasvith) [inline internal/arch/command-table.go]: we can use 0 I guess, We can not have minimum infinite args, so its zero when it requires minimum 0 args. Sorry about introducing min as -1. Also `min(len) = 0` which i forgot 🗡

**Comment by kasvith** [comment]
Seems testify is missing from deps.

**Comment by kasvith** [comment]
https://github.com/golang/dep/issues/854 we need to manually install it.

**Comment by Chyroc** [comment]
yes, i am makeing a pr to do it.

**Comment by kasvith** [comment]
Use master branch as base then it can be easily merged...

File: internal/arch/command-table.go [hunk 3]
```diff
@@ -35,21 +35,23 @@ type CommandFunc func(*db.DB, []string) *protcl.Message
 type Command struct {
 	ModifyKeySpace bool
 	Fn             CommandFunc
+	MinArgs        int // 0 ~ +inf
```
[the remark this answers is not in the document]
Comment (kasvith) [reply internal/arch/command-table.go]: Yah

**Comment by kasvith** [comment]
Nice job

**Pull request merged**